src/ad-cid.js

```diff
@@ -37,10 +38,13 @@ export function getAdCid(adElement) {
  * @param {!./service/ampdoc-impl.AmpDoc|!Node} ampDoc
  * @param {string} clientIdScope
  * @param {string=} opt_clientIdCookieName
+ * @param {number=} opt_timeout
  * @return {!Promise<string|undefined>} A promise for a CID or undefined.
  */
 export function getOrCreateAdCid(
-  ampDoc, clientIdScope, opt_clientIdCookieName) {
+  ampDoc, clientIdScope, opt_clientIdCookieName, opt_timeout) {
+  const timeout = opt_timeout != undefined && isFiniteNumber(opt_timeout) ?
```

Comment:
Instead of finiteNumber, can you just round up to nearest whole number and consider adding warn?  Otherwise the number is just not honored.  Also are you positive the typing is passed correctly given this is parsed from a string?  Perhaps parseInt is sufficient which will cut the decimals?

Reply:
Changed so we pass it in as a parsedInt.